File: day08.cpp
#include <iostream>
using namespace std;

bool isBinary(string str)
{
    for (int i = 0; i < str.length(); i++)
    {
        if (str[i] != '1' && str[i] != '0')
        {
            return false;
        }
    }
    return true;
}

int main()
{
    string str;
    cout << "Enter the string to validate the it is binary string or not : ";
    cin >> str;

    cout << (isBinary(str) ? "Yes it is a binary string" : "No it is not a binary string");

    return 0;
}

File: day08.java
import java.util.Scanner;

public class day08 {
    public static void main(String[] args) {

        String str;
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the string to validate it is the binary string or not : ");
        str = scanner.nextLine();
        scanner.close();

        if (isBinary(str)) {
            System.out.println("yes it is binary string");
        } else {
            System.out.println("No it is not binary string");
        }
    }

    public static boolean isBinary(String str) {
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) != '1' && str.charAt(i) != '0') {
                return false;
            }
        }

        return true;
    }
}
